capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java

```diff
@@ -40,61 +44,105 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
+      String sStackTrace = sw.toString();
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, sStackTrace);
       return;
     }
     response.setStatus(HttpServletResponse.SC_OK);
     response.setContentType("application/json;");
   }
 
-  public void sendEmail(String email, String html) throws MessagingException {
-    // Recipient's email ID needs to be mentioned.
-    String to = email;
-    // Sender's email ID needs to be mentioned
-    String from = "shopsmart.step@gmail.com";
-    // Assuming you are sending email from through gmails smtp
-    String host = "smtp.gmail.com";
-    // Get system properties
-    Properties properties = System.getProperties();
-    // Setup mail server
-    properties.put("mail.smtp.host", host);
-    properties.put("mail.smtp.port", "465");
-    properties.put("mail.smtp.ssl.enable", "true");
-    properties.put("mail.smtp.auth", "true");
-    // Get the Session object.// and pass username and password
-    Session session =
-        Session.getInstance(
-            properties,
-            new javax.mail.Authenticator() {
-              protected PasswordAuthentication getPasswordAuthentication() {
-                return new PasswordAuthentication(
-                    GmailCredentials.getUsername(), GmailCredentials.getPassword());
-              }
-            });
-    // Used to debug SMTP issues
-    session.setDebug(true);
+  public void sendEmail(String email, String storesHTML) throws Exception {
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, null);
     // Create a default MimeMessage object.
-    MimeMessage message = new MimeMessage(session);
+    Message message = new MimeMessage(session);
     // Set From: header field of the header.
-    message.setFrom(new InternetAddress(from));
+    message.setFrom(new InternetAddress("shopsmart@step39-2020.appspotmail.com", "Shopsmart"));
     // Set To: header field of the header.
-    message.addRecipient(Message.RecipientType.TO, new InternetAddress(to));
+    message.addRecipient(Message.RecipientType.TO, new InternetAddress(email));
     // Set Subject: header field
     message.setSubject("Message From ShopSmart");
     // Now set the actual message
-    message.setContent(html, "text/html");
-    System.out.println("sending...");
+    message.setContent(storesHTML, "text/html");
     // Send message
     Transport.send(message);
   }
 
+  private String getStoresHTML(
+      List<Store> stores, List<String> itemTypes, double latitude, double longitude) {
+    String url = "https://step39-2020.uc.r.appspot.com/";
+    try {
+      URIBuilder urlb = new URIBuilder("https://step39-2020.uc.r.appspot.com/stores/");
+      for (String item : itemTypes) {
+        urlb.addParameter("items", item);
+      }
+      urlb.addParameter("latitude", "" + latitude);
+      urlb.addParameter("longitude", "" + longitude);
+      urlb.addParameter("distanceValue", "2147483647");
```

Comment:
what is this distance value?

Reply:
It is the value for preferred distance when we allowed users to set their preference before, but now we just automatically set it to the max apparently